src/unit_tests/os_auth/test_auth_validate.c

```diff
@@ -257,9 +272,76 @@ static void test_w_auth_validate_groups(void **state) {
     err = w_auth_validate_groups(EXISTENT_GROUP1","EXISTENT_GROUP2","UNKNOWN_GROUP, response);
     assert_int_equal(err, OS_INVALID);
     assert_string_equal(response, "ERROR: Invalid group: "UNKNOWN_GROUP"");
+}
+
+static void test_w_auth_replace_agent_force_disabled(void **state) {
+    w_err_t err;
+    keyentry key;
+    key.id = AGENT1_ID;
+    os_calloc(1, sizeof(os_ip), key.ip);
+    key.ip->ip = NEW_IP1;
+    key.name = NEW_AGENT1;
+
+    expect_string(__wrap__minfo, formatted_msg, "Agent '001' won´t be removed because the force option is disabled.");
+    err = w_auth_replace_agent(&key, NULL, &config.force_options);
+
+    assert_int_equal(err, OS_INVALID);
+    os_free(key.ip);
+}
+
+static void test_w_auth_replace_agent_not_comply_antiquity(void **state) {
+    w_err_t err;
+    keyentry key;
+    key.id = AGENT1_ID;
+    os_calloc(1, sizeof(os_ip), key.ip);
+    key.ip->ip = NEW_IP1;
+    key.name = NEW_AGENT1;
+
+    // Mocking antiquity
+    will_return(__wrap_OS_AgentAntiquity, 10);
+    config.force_options.connection_time = 100;
 
+    expect_string(__wrap__minfo, formatted_msg, "Agent '001' doesn´t comply with the antiquity to be removed.");
+    err = w_auth_replace_agent(&key, NULL, &config.force_options);
+
+    assert_int_equal(err, OS_INVALID);
+    os_free(key.ip);
+    config.force_options.connection_time = 0;
 }
 
+static void test_w_auth_replace_agent_existent_key_hash(void **state) {
+    w_err_t err;
+    keyentry key;
+    key.id = AGENT1_ID;
+    os_calloc(1, sizeof(os_ip), key.ip);
+    key.ip->ip = NEW_IP1;
+    key.name = NEW_AGENT1;
+    key.raw_key = "1234";
+    char *key_hash = "15153d246b71789195b48778875af94f9378ecf9";
```

Comment:
We should add a comment here explaining that "**15153d246b71789195b48778875af94f9378ecf9**" is the precalculated **os_sha1** of **id** + **name** + **raw_key**

Reply:
Done!